sgx-traits/src/lib.rs

```diff
@@ -15,40 +15,30 @@
 use std::io::Result;
 
 use bitflags::bitflags;
-use sgx_types::page::{Flags as PageFlags, SecInfo};
-use sgx_types::{secs::Secs, sig::Signature, tcs::Tcs, Offset};
+use sgx_types::{page, secs, sig};
 
 bitflags! {
     pub struct Flags: u64 {
         const MEASURE = 1 << 0;
     }
 }
 
-pub trait Enclave<'e> {
-    unsafe fn enter(&self, offset: &mut Offset<Tcs>) -> Result<()>;
+pub trait Enclave {
+    unsafe fn enter(&self, offset: usize) -> Result<()>;
 }
 
-pub trait Builder<'b>: Sized {
-    type Enclave: Enclave<'b>;
+pub trait Builder: Sized {
+    type Enclave: Enclave;
 
-    fn new(secs: Secs) -> Result<Self>;
+    fn new(secs: secs::Secs) -> Result<Self>;
 
-    unsafe fn add_tcs(&mut self, tcs: Tcs, offset: usize) -> Result<Offset<Tcs>>;
-
-    unsafe fn add_struct<T>(
+    unsafe fn add(
```

Comment:
This seems to replace the functionality of `add_tcs`, `add_struct`, and `add_slice`, is that right?

Reply:
Correct.